solr/solrj/src/java/org/apache/solr/client/solrj/response/QueryResponse.java

```diff
@@ -286,10 +286,10 @@ private void extractGroupedInfo( NamedList<Object> info ) {
 
           _groupResponse.add(groupedCommand);
         } else if (queryCommand != null) {
-          Integer iMatches = (Integer) oMatches;
+          Integer iMatches = ((Number) oMatches).intValue();
           GroupCommand groupCommand;
           if (oNGroups != null) {
-            Integer iNGroups = (Integer) oNGroups;
+            Integer iNGroups = ((Number) oNGroups).intValue();
```

Comment:
I see that these `Integer`s would get passed as `int` later on. So the previous code would NPE if `oMatches` or `oNGroups` was null. So this behavior would be preserved. So no need for null checking here :+1: .